@ChristophMajcenAtXxxlutz , looks like there's a mypy error now:

> Running mypy type checking ...02:08
src/confluent_kafka/schema_registry/common/avro.py:270: error:(B Unsupported right operand type for in ((B"Any | None"(B)  (B[operator](B02:12
Found 1 error in 1 file (checked 106 source files)(B

--- a/src/confluent_kafka/schema_registry/common/avro.py
+++ b/src/confluent_kafka/schema_registry/common/avro.py
@@ -267,7 +267,7 @@ def _collapse_schema(schema: AvroSchema, encountered_references=None) -> AvroSch
         elif schema_type == "record":
             name = schema.get("name")
             namespace = schema.get("namespace")
-            full_name = name if namespace is None or '.' in name else f"{namespace}.{name}"
+            full_name = name if namespace is None or (name is not None and '.' in name) else f"{namespace}.{name}"
             if full_name in encountered_references:
                 return schema["name"]
             encountered_references.append(full_name)

